### PR TITLE
Fix: Request clones time-dimensions correctly

### PR DESCRIPTION
**Pull request by cythrawll**
## Description

 When cloning a report, time-dimensions get cloned as undefined

## Proposed Changes

- Make sure dimension/time-dimension metadata gets queried as the right type during clone
- Also gradle update for java > 8

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/core/addon/models/bard-request/request.js [hunk 1]
```diff
@@ -551,7 +551,9 @@ export default Fragment.extend(Validations, {
   clone() {
     let clonedRequest = this.toJSON(),
       store = this.store,
-      metadataService = this.metadataService;
+      metadataService = this.metadataService,
+      dimensionTypeMap = Object.fromEntries(this.dimensions.map(dim => [dim.dimension.id, dim.dimension.valueType])),
+      filterTypeMap = Object.fromEntries(this.filters.map(filter => [filter.dimension.id, filter.dimension.valueType]));
```
Comment (kevinhinterlong) [inline packages/core/addon/models/bard-request/request.js:556]: can we make each line its own assignment
```js
const store = this.store;
const metadataService = this.metadataService;
// etc
```

File: packages/core/addon/models/bard-request/request.js [hunk 2]
```diff
@@ -551,7 +551,9 @@ export default Fragment.extend(Validations, {
   clone() {
     let clonedRequest = this.toJSON(),
       store = this.store,
-      metadataService = this.metadataService;
+      metadataService = this.metadataService,
+      dimensionTypeMap = Object.fromEntries(this.dimensions.map(dim => [dim.dimension.id, dim.dimension.valueType])),
```
Comment (kevinhinterlong) [inline packages/core/addon/models/bard-request/request.js:555]: `Object.fromEntries` 

nice

File: packages/core/addon/models/bard-request/request.js [hunk 3]
```diff
@@ -574,7 +580,11 @@ export default Fragment.extend(Validations, {
 
       filters: clonedRequest.filters.map(filter =>
         store.createFragment('bard-request/fragments/filter', {
-          dimension: metadataService.getById('dimension', filter.dimension, clonedRequest.dataSource),
+          dimension: metadataService.getById(
+            dimensionTypeMap[filter.dimension] === 'date' ? 'time-dimension' : 'dimension',
```
Comment (kevinhinterlong) [inline packages/core/addon/models/bard-request/request.js:584]:
```suggestion
            filterTypeMap[filter.dimension] === 'date' ? 'time-dimension' : 'dimension',
```
?

File: packages/core/addon/models/bard-request/request.js (hunk 2 above)
Comment (cythrawll) [reply packages/core/addon/models/bard-request/request.js:555]: I know right? so much better than `reduce` I been waiting for the time I needed to make an object from an array for this 😄

**Comment by cythrawll** [comment]
Easy enough.

**Pull request merged**